File: AnyTorah/Models/HebrewUtils.swift
import Foundation

extension String {
    /// Strip Hebrew nikud (vowel points U+05B0–U+05C7) and cantillation marks (U+0591–U+05AF).
    var strippingNikud: String {
        String(String.UnicodeScalarView(unicodeScalars.filter { $0.value < 0x0591 || $0.value > 0x05C7 }))
    }

    var strippingSeferPrefix: String {
        let prefix = "ספר "
        return hasPrefix(prefix) ? String(dropFirst(prefix.count)) : self
    }
}

File: AnyTorahWeb/lib/hebrewUtils.ts
// Ported from AnyTorah/AnyTorah/Models/HebrewUtils.swift (`String.strippingNikud`).

/** Strips Hebrew cantillation marks and vowel points (U+0591–U+05C7), matching native's
 *  `strippingNikud`. Used for saHebrewMode display names, which are stored with full nikud. */
export function stripNikud(s: string): string {
  return Array.from(s)
    .filter((ch) => {
      const code = ch.codePointAt(0)!;
      return code < 0x0591 || code > 0x05c7;
    })
    .join("");
}
